source/cppfront.cpp

```diff
@@ -4684,40 +4752,28 @@ class cppfront
                             + "_as_base";
                     }
 
-                    //  Flush any 'out' parameter initializations
-                    auto out_inits_with_commas = [&]() -> std::string {
-                        auto ret = std::string{};
-                        for (auto& init : out_inits) {
-                            ret += init + ", ";
+                    //  If there are any pending statements, wedge them into
+                    //  this initializer using an immediately invoked lambda
+                    auto& mem_inits = current_functions.back().prolog.mem_inits;
+                    if (!pending_statements.empty())
+                    {
+                        auto object_type = print_to_string( *(*object)->get_object_type() );
+                        mem_inits.push_back(separator + object_name + "{ [&]() -> " + object_type + " {");
```

Comment:
Fixing the return type to the type of the data member to initialize is NG.
This should achieve the semantics one would expected reading the Cpp2 code:
```suggestion
                        mem_inits.push_back(separator + object_name + "{ [&]() -> decltype(auto) {");
```